core/src/actors/peers_manager.rs

```diff
@@ -20,6 +25,42 @@ pub struct PeersManager {
     peers: Peers,
 }
 
+impl PeersManager {
+    /// Method to periodically dump peers into storage
+    fn storage_peers(&self, ctx: &mut Context<Self>, storage_peers_period: u64) {
+        // Schedule the discovery_peers with a given period
+        ctx.run_later(
+            Duration::from_secs(storage_peers_period),
+            move |act, ctx| {
+                // Get storage manager address
+                let storage_manager_addr = System::current().registry().get::<StorageManager>();
+
+                // Dump peers into storage. `AsyncContext::wait` register
+                // future within context, but context waits until this future resolves
+                // before processing any other events.
+                storage_manager_addr
+                    .send(Put::from_value(PEERS_KEY, &act.peers).unwrap())
+                    .into_actor(act)
+                    .then(|res, _act, ctx| {
+                        match res {
+                            Ok(Ok(_)) => debug!("Peers manager successfully dump peers to storage"),
+                            _ => {
+                                debug!("Peers manager dump peers to storage failed");
+                                // FIXME(#72): a full stop of the session is not correct (unregister should
+                                // be skipped)
+                                ctx.stop()
```

Comment:
The `PeersManager` should not stop itself because of an error in the `StorageManager` 
```suggestion
```

Reply:
done!!